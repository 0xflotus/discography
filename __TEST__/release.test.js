import testSnapshot from './fixtures/testSnapshot';

test('Should have all basic fields on the release type', async () => {
  const query = `
    query {
      lookup {
        release (id: 2) {
          id
          title
          country
          released
          genres
          styles
          formats {
            position
            formatName
            quantity
          }
          images {
            type
            height
            width
          }
          master {
            id
          }
        }
      }
    }
  `;

  await testSnapshot(query);
});

test('Should implement the release artist connection', async () => {
  const query = `
    query {
      lookup {
        release (id: 1) {
          id
          artists (first: 1) {
            edges {
              node {
                id
              }
            }
          }
        }
      }
    }
  `;

  await testSnapshot(query);
});

test('Should implement the release label connection', async () => {
  const query = `
    query {
      lookup {
        release (id: 1) {
          id
          labels (first: 1) {
            edges {
              node {
                id
              }
            }
          }
        }
      }
    }
  `;

  await testSnapshot(query);
});

test('Should implement the release track connection', async () => {
  const query = `
    query {
      lookup {
        release (id: 1) {
          id
          tracks (first: 1) {
            edges {
              node {
                trackno
                title
              }
            }
          }
        }
      }
    }
  `;

  await testSnapshot(query);
});

test('Should implement the extra artist connection', async () => {
  const query = `
    query {
      lookup {
        release (id: 2) {
          id
          extraArtists (first: 1) {
            edges {
              node {
                id
              }
            }
          }
        }
      }
    }
  `;

  await testSnapshot(query);
});
