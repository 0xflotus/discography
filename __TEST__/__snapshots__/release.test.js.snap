// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should have all basic fields on the release type 1`] = `
Object {
  "data": Object {
    "lookup": Object {
      "release": Object {
        "country": "Sweden",
        "formats": Array [
          Object {
            "formatName": "Vinyl",
            "position": 1,
            "quantity": 1,
          },
        ],
        "genres": Array [
          "Electronic",
        ],
        "id": 2,
        "images": Array [
          Object {
            "height": 394,
            "type": "primary",
            "width": 400,
          },
          Object {
            "height": 600,
            "type": "secondary",
            "width": 600,
          },
        ],
        "master": Object {
          "id": 713738,
        },
        "released": "1998-06-00",
        "styles": Array [
          "Broken Beat",
          "Techno",
          "Tech House",
        ],
        "title": "Knockin' Boots Vol 2 Of 2",
      },
    },
  },
}
`;

exports[`Should implement the extra artist connection 1`] = `
Object {
  "data": Object {
    "lookup": Object {
      "release": Object {
        "extraArtists": Object {
          "edges": Array [
            Object {
              "node": Object {
                "id": 26,
              },
            },
          ],
        },
        "id": 2,
      },
    },
  },
}
`;

exports[`Should implement the release artist connection 1`] = `
Object {
  "data": Object {
    "lookup": Object {
      "release": Object {
        "artists": Object {
          "edges": Array [
            Object {
              "node": Object {
                "id": 1,
              },
            },
          ],
        },
        "id": 1,
      },
    },
  },
}
`;

exports[`Should implement the release label connection 1`] = `
Object {
  "data": Object {
    "lookup": Object {
      "release": Object {
        "id": 1,
        "labels": Object {
          "edges": Array [
            Object {
              "node": Object {
                "id": 5,
              },
            },
          ],
        },
      },
    },
  },
}
`;

exports[`Should implement the release track connection 1`] = `
Object {
  "data": Object {
    "lookup": Object {
      "release": Object {
        "id": 1,
        "tracks": Object {
          "edges": Array [
            Object {
              "node": Object {
                "title": "Ã–stermalm",
                "trackno": 1,
              },
            },
          ],
        },
      },
    },
  },
}
`;
